test/support/warning_filter.rb

```diff
@@ -0,0 +1,53 @@
+require 'pathname'
+require 'set'
+
+module RegexpParserTest
+  # Stream filter for warnings
+  class WarningFilter
```

Comment:
`WarningObserver` is probably better since all it is doing is capturing warnings emitted by ruby.